src/sentry/static/sentry/app/components/hovercard.jsx

```diff
@@ -1,49 +1,99 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
+import intersection from 'lodash/intersection';
 
 class Hovercard extends React.Component {
   static propTypes = {
+    displayTimeout: PropTypes.number,
     className: PropTypes.string,
     containerClassName: PropTypes.string,
     header: PropTypes.node,
     body: PropTypes.node,
   };
 
+  static defaultProps = {
+    displayTimeout: 100,
+  };
+
   constructor(...args) {
     super(...args);
     this.state = {
       visible: false,
+      rendered: false,
     };
+    this.hoverWait = null;
+    this.cardElement = null;
   }
 
-  handleToggleHovercard = () => {
-    let {header, body} = this.props;
+  handleToggleOn = () => this.toggleHovercard(true);
+  handleToggleOff = () => this.toggleHovercard(false);
+
+  toggleHovercard = visible => {
+    const {header, body} = this.props;
 
     // Don't toggle hovercard if both of these are null
     if (!header && !body) return;
 
-    this.setState({
-      visible: !this.state.visible,
-    });
+    if (this.hoverWait !== null) {
+      clearTimeout(this.hoverWait);
+    }
+
+    const rendered = visible;
+    const timeout = this.props.displayTimeout;
+
+    // Immediately render the hovercard, but don't mark it as visible until
+    // after the delay period. This allows us to compute the size of the
+    // hovercard to position it before it is made visible.
+    if (visible) {
+      this.setState({rendered});
+    }
+
+    this.hoverWait = setTimeout(() => this.setState({visible, rendered}), timeout);
   };
 
+  positionClasses() {
+    if (!this.cardElement || !this.state.visible) return {};
```

Comment:
this could also be tested by mocking boundingClientRect but maybe not worthwhile